doc/man/man3/Config.3
.TH "Config" 3AZURE2" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Config \- A configuration structure for AZURE\&.  

.SH SYNOPSIS
.br
.PP
.PP
\fC#include <Config\&.h>\fP
.SS "Public Types"

.in +1c
.ti -1c
.RI "enum \fBParameterFlags\fP { \fBUSE_AMATRIX\fP =(1<<0), \fBPERFORM_ERROR_ANALYSIS\fP =(1<<1), \fBPERFORM_FIT\fP =(1<<2), \fBCALCULATE_WITH_DATA\fP =(1<<3), \fBUSE_PREVIOUS_PARAMETERS\fP =(1<<4), \fBUSE_EXTERNAL_CAPTURE\fP =(1<<5), \fBUSE_PREVIOUS_INTEGRALS\fP =(1<<6), \fBCALCULATE_REACTION_RATE\fP =(1<<7), \fBTRANSFORM_PARAMETERS\fP =(1<<8), \fBUSE_BRUNE_FORMALISM\fP =(1<<9), \fBIGNORE_ZERO_WIDTHS\fP =(1<<10), \fBUSE_RMC_FORMALISM\fP =(1<<11), \fBUSE_GSL_COULOMB_FUNC\fP =(1<<12), \fBUSE_LONGWAVELENGTH_APPROX\fP =(1<<13) }"
.br
.ti -1c
.RI "enum \fBCheckFileFlags\fP { \fBCHECK_COMPOUND_NUCLEUS\fP =(1<<0), \fBCHECK_PATHWAYS\fP =(1<<1), \fBCHECK_DATA\fP =(1<<2), \fBCHECK_ENERGY_DEP\fP =(1<<3), \fBCHECK_LEGENDRE\fP =(1<<4), \fBCHECK_BOUNDARY_CONDITIONS\fP =(1<<5), \fBCHECK_ANGULAR_DISTS\fP =(1<<6), \fBCHECK_COUL_AMPLITUDES\fP =(1<<7) }"
.br
.in -1c
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBConfig\fP (std::ostream &stream)"
.br
.ti -1c
.RI "void \fBReset\fP ()"
.br
.ti -1c
.RI "int \fBReadConfigFile\fP ()"
.br
.ti -1c
.RI "int \fBCheckForInputFiles\fP ()"
.br
.in -1c
.SS "Public Attributes"

.in +1c
.ti -1c
.RI "std::ostream & \fBoutStream\fP"
.br
.RI "Output stream\&. "
.ti -1c
.RI "std::string \fBconfigfile\fP"
.br
.RI "The runtime configuration file name\&. "
.ti -1c
.RI "bool \fBstopFlag\fP"
.br
.RI "A control variable to stop AZURE calculation\&. "
.ti -1c
.RI "unsigned int \fBparamMask\fP"
.br
.RI "A bitmask for the encoding of configuration flags\&. "
.ti -1c
.RI "unsigned int \fBscreenCheckMask\fP"
.br
.RI "A bitmask storing which checks are printed to screen\&. "
.ti -1c
.RI "unsigned int \fBfileCheckMask\fP"
.br
.RI "A bitmask storing which checks are printed to file\&. "
.ti -1c
.RI "double \fBchiVariance\fP"
.br
.RI "If performError is true, sets the value of Up (the acceptable variance from the minimum chi-squared\&. "
.ti -1c
.RI "std::string \fBoutputdir\fP"
.br
.RI "The path of the output directory\&. "
.ti -1c
.RI "std::string \fBcheckdir\fP"
.br
.RI "The path of the check files directory\&. "
.ti -1c
.RI "std::string \fBparamfile\fP"
.br
.RI "The name of the parameters file from which to read\&. "
.ti -1c
.RI "std::string \fBintegralsfile\fP"
.br
.RI "The name of the external capture amplitudes file from which to read\&. "
.ti -1c
.RI "\fBRateParams\fP \fBrateParams\fP"
.br
.RI "Parameters for calculating reaction rate\&. "
.in -1c
.SS "Static Public Attributes"

.in +1c
.ti -1c
.RI "static const int \fBmaxLOrder\fP =20"
.br
.RI "A constant indicating the maximum order of the Legendre polynomials to calculate\&. "
.in -1c
.SH "Detailed Description"
.PP 
A configuration structure for AZURE\&. 

The configuration structure is created from the runtime file passed to the AZURE executable, as well as the options specified in the command shell prompt\&. 
.PP
Definition at line \fB37\fP of file \fBConfig\&.h\fP\&.
.SH "Member Enumeration Documentation"
.PP 
.SS "enum \fBConfig::CheckFileFlags\fP"
Bit flags for check file control in AZURE2\&. 
.PP
\fBEnumerator\fP
.in +1c
.TP
\fB\fICHECK_COMPOUND_NUCLEUS \fP\fP
.TP
\fB\fICHECK_PATHWAYS \fP\fP
.TP
\fB\fICHECK_DATA \fP\fP
.TP
\fB\fICHECK_ENERGY_DEP \fP\fP
.TP
\fB\fICHECK_LEGENDRE \fP\fP
.TP
\fB\fICHECK_BOUNDARY_CONDITIONS \fP\fP
.TP
\fB\fICHECK_ANGULAR_DISTS \fP\fP
.TP
\fB\fICHECK_COUL_AMPLITUDES \fP\fP
.PP
Definition at line \fB63\fP of file \fBConfig\&.h\fP\&.
.SS "enum \fBConfig::ParameterFlags\fP"
Bit flags for various options in AZURE2\&. 
.PP
\fBEnumerator\fP
.in +1c
.TP
\fB\fIUSE_AMATRIX \fP\fP
.TP
\fB\fIPERFORM_ERROR_ANALYSIS \fP\fP
.TP
\fB\fIPERFORM_FIT \fP\fP
.TP
\fB\fICALCULATE_WITH_DATA \fP\fP
.TP
\fB\fIUSE_PREVIOUS_PARAMETERS \fP\fP
.TP
\fB\fIUSE_EXTERNAL_CAPTURE \fP\fP
.TP
\fB\fIUSE_PREVIOUS_INTEGRALS \fP\fP
.TP
\fB\fICALCULATE_REACTION_RATE \fP\fP
.TP
\fB\fITRANSFORM_PARAMETERS \fP\fP
.TP
\fB\fIUSE_BRUNE_FORMALISM \fP\fP
.TP
\fB\fIIGNORE_ZERO_WIDTHS \fP\fP
.TP
\fB\fIUSE_RMC_FORMALISM \fP\fP
.TP
\fB\fIUSE_GSL_COULOMB_FUNC \fP\fP
.TP
\fB\fIUSE_LONGWAVELENGTH_APPROX \fP\fP
.PP
Definition at line \fB44\fP of file \fBConfig\&.h\fP\&.
.SH "Constructor & Destructor Documentation"
.PP 
.SS "Config::Config (std::ostream & stream)"
The constructor of the \fBConfig\fP class sets defaults and the stream reference for output\&. 
.PP
Definition at line \fB12\fP of file \fBConfig\&.cpp\fP\&.
.SH "Member Function Documentation"
.PP 
.SS "int Config::CheckForInputFiles ()"
If stat() is enabled, this function checks for the output and checks directories at runtime\&. 
.PP
Definition at line \fB102\fP of file \fBConfig\&.cpp\fP\&.
.SS "int Config::ReadConfigFile ()"
This funciton reads the configuration file and parses various options\&. 
.PP
Definition at line \fB35\fP of file \fBConfig\&.cpp\fP\&.
.SS "void Config::Reset ()"
This function resets \fBConfig\fP structure\&. 
.PP
Definition at line \fB20\fP of file \fBConfig\&.cpp\fP\&.
.SH "Member Data Documentation"
.PP 
.SS "std::string Config::checkdir"

.PP
The path of the check files directory\&. 
.PP
Definition at line \fB90\fP of file \fBConfig\&.h\fP\&.
.SS "double Config::chiVariance"

.PP
If performError is true, sets the value of Up (the acceptable variance from the minimum chi-squared\&. 
.PP
Definition at line \fB86\fP of file \fBConfig\&.h\fP\&.
.SS "std::string Config::configfile"

.PP
The runtime configuration file name\&. 
.PP
Definition at line \fB76\fP of file \fBConfig\&.h\fP\&.
.SS "unsigned int Config::fileCheckMask"

.PP
A bitmask storing which checks are printed to file\&. 
.PP
Definition at line \fB84\fP of file \fBConfig\&.h\fP\&.
.SS "std::string Config::integralsfile"

.PP
The name of the external capture amplitudes file from which to read\&. 
.PP
Definition at line \fB94\fP of file \fBConfig\&.h\fP\&.
.SS "const int Config::maxLOrder =20\fC [static]\fP"

.PP
A constant indicating the maximum order of the Legendre polynomials to calculate\&. 
.PP
Definition at line \fB98\fP of file \fBConfig\&.h\fP\&.
.SS "std::string Config::outputdir"

.PP
The path of the output directory\&. 
.PP
Definition at line \fB88\fP of file \fBConfig\&.h\fP\&.
.SS "std::ostream& Config::outStream"

.PP
Output stream\&. 
.PP
Definition at line \fB74\fP of file \fBConfig\&.h\fP\&.
.SS "std::string Config::paramfile"

.PP
The name of the parameters file from which to read\&. 
.PP
Definition at line \fB92\fP of file \fBConfig\&.h\fP\&.
.SS "unsigned int Config::paramMask"

.PP
A bitmask for the encoding of configuration flags\&. 
.PP
Definition at line \fB80\fP of file \fBConfig\&.h\fP\&.
.SS "\fBRateParams\fP Config::rateParams"

.PP
Parameters for calculating reaction rate\&. 
.PP
Definition at line \fB96\fP of file \fBConfig\&.h\fP\&.
.SS "unsigned int Config::screenCheckMask"

.PP
A bitmask storing which checks are printed to screen\&. 
.PP
Definition at line \fB82\fP of file \fBConfig\&.h\fP\&.
.SS "bool Config::stopFlag"

.PP
A control variable to stop AZURE calculation\&. 
.PP
Definition at line \fB78\fP of file \fBConfig\&.h\fP\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for AZURE2 from the source code\&.


src/Config.cpp
#include "Config.h"
#ifndef NO_STAT
#include <sys/stat.h>
#endif
#include <iostream>

/*!
 * The constructor of the Config class sets defaults and the 
 * stream reference for output.
 */

Config::Config(std::ostream& stream) : outStream(stream) {
  Reset();
}

/*!
 * This function resets Config structure.
 */

void Config::Reset() {
  chiVariance=1.0;
  screenCheckMask=0;
  fileCheckMask=0;
  paramMask=0;
  paramMask |= (USE_AMATRIX|USE_BRUNE_FORMALISM|IGNORE_ZERO_WIDTHS|TRANSFORM_PARAMETERS|CALCULATE_WITH_DATA|USE_LONGWAVELENGTH_APPROX);
  stopFlag=false;
  outputdir="";
  checkdir="";
}

/*!
 * This funciton reads the configuration file and parses various options.
 */

int Config::ReadConfigFile() {
  std::string dummy; std::string temp;
  std::ifstream in(configfile.c_str());
  if(!in) return -1;
  std::string line="";
  while(line!="<config>"&&!in.eof()) getline(in,line);
  if(line!="<config>") return -1;
  in >> temp;getline(in,dummy);
  if(temp=="true") paramMask |= USE_AMATRIX;
  else paramMask &= ~USE_AMATRIX;
  getline(in,dummy);
  int poundSignPos=dummy.find_last_of('#');
  if(poundSignPos==std::string::npos) temp=dummy;
  else temp=dummy.substr(0,poundSignPos);
  int p2 = temp.find_last_not_of(" \n\t\r");
  if (p2 != std::string::npos) {  
    int p1 = temp.find_first_not_of(" \n\t\r");
    if (p1 == std::string::npos) p1 = 0;
    outputdir=temp.substr(p1,(p2-p1)+1);
  } else outputdir=std::string();  
  getline(in,dummy);
  poundSignPos=dummy.find_last_of('#');
  if(poundSignPos==std::string::npos) temp=dummy;
  else temp=dummy.substr(0,poundSignPos);
  p2 = temp.find_last_not_of(" \n\t\r");
  if (p2 != std::string::npos) {  
    int p1 = temp.find_first_not_of(" \n\t\r");
    if (p1 == std::string::npos) p1 = 0;
    checkdir=temp.substr(p1,(p2-p1)+1);
  } else checkdir=std::string();   
  in >> temp;getline(in,dummy);
  if(temp=="screen") screenCheckMask |= CHECK_COMPOUND_NUCLEUS;
  else if(temp=="file") fileCheckMask |= CHECK_COMPOUND_NUCLEUS;
  in >> temp;getline(in,dummy);
  if(temp=="screen") screenCheckMask |= CHECK_BOUNDARY_CONDITIONS;
  else if(temp=="file") fileCheckMask |= CHECK_BOUNDARY_CONDITIONS;
  in >> temp;getline(in,dummy);
  if(temp=="screen") screenCheckMask |= CHECK_DATA;
  else if(temp=="file") fileCheckMask |= CHECK_DATA;
  in >> temp;getline(in,dummy);
  if(temp=="screen") screenCheckMask |= CHECK_ENERGY_DEP;
  else if(temp=="file") fileCheckMask |= CHECK_ENERGY_DEP;
  in >> temp;getline(in,dummy);
  if(temp=="screen") screenCheckMask |= CHECK_LEGENDRE;
  else if(temp=="file") fileCheckMask |= CHECK_LEGENDRE;
  in >> temp;getline(in,dummy);
  if(temp=="screen") screenCheckMask |= CHECK_COUL_AMPLITUDES;
  else if(temp=="file") fileCheckMask |= CHECK_COUL_AMPLITUDES;
  in >> temp;getline(in,dummy);
  if(temp=="screen") screenCheckMask |= CHECK_PATHWAYS;
  else if(temp=="file") fileCheckMask |= CHECK_PATHWAYS;
  in >> temp;getline(in,dummy);
  if(temp=="screen") screenCheckMask |= CHECK_ANGULAR_DISTS;
  else if(temp=="file") fileCheckMask |= CHECK_ANGULAR_DISTS;
  line="";
  while(line!="</config>"&&!in.eof()) getline(in,line);
  if(line!="</config>") return -1;
  in.close();
  return 0;
}

/*!
 * If stat() is enabled, this function checks for the output and checks 
 * directories at runtime.
 */

#ifndef NO_STAT
int Config::CheckForInputFiles() {
  struct stat buffer;
  if(stat(outputdir.c_str(),&buffer) != 0) {
    outStream << "Could not find output directory: " << outputdir << ". Check that it exists." << std::endl;
    return -1;
  }
  if(stat(checkdir.c_str(),&buffer) != 0) {
    outStream << "Could not find checks directory: " << checkdir << ". Check that it exists." << std::endl;
    return -1;
  }
  return 0;
}
#endif
